fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -434,7 +434,12 @@ public Response updateSparql(@ContentLocation final InputStream requestBodyStrea
             }
             session.save();
 
-            addCacheControlHeaders(servletResponse, resource(), session);
+            if (resource() instanceof FedoraBinary) {
+                final NonRdfSourceDescription description = ((FedoraBinary)resource()).getDescription();
```

Comment:
Do stanzas like this become easier to understand if we move `getDescription()` to `FedoraResource` and let an RDF resource describe itself?